src/components/StatedBalanceForm.test.tsx

```diff
@@ -0,0 +1,59 @@
+import Enzyme, {mount, shallow} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import React, {ReactElement} from "react";
+import StatedBalanceForm from "./StatedBalanceForm";
+import {saveValues} from "../core/Memory";
+
+
+Enzyme.configure({adapter: new Adapter()});
+
+describe('StatedBalanceForm', () => {
+    it('renders without crashing given the required props', () => {
+        const save_value = jest.fn();
+        const get_value = jest.fn();
+        const wrapper = shallow(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
+        expect((wrapper)).toMatchSnapshot();
+    })
+})
+
+test("When user inputs, then get_values is called ", () => {
```

Comment:
You can wrap all these tests in describe("Given StatedBalanceForm")

Reply:
'When user submits balance form' vs "Given StatedBalanceForm"? Which is better and why? I think both are pretty similar.

Reply:
describe can contain more than one test.
It is just an aggregate of tests that describes the groups of them.
Fully described test should be constructed as "given ... when ... then ...", however we tend to skip some parts.